### PR TITLE
Polish InferReshape documentation

### PR DESCRIPTION
**Pull request by psyyz10**
## What changes were proposed in this pull request?

Polish InferReshape documentation

### CODE REVIEW

**Comment by SeaOfOcean** [comment]
LGTM

**Comment by SeaOfOcean** [comment]
update corresponding python doc

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala [hunk 1]
```diff
@@ -23,21 +23,27 @@ import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
...
- * Reshape with the support of infered size,
- * Positive numbers are used directly, setting the corresponding dimension of the output tensor.
- * In addition, two special values are accepted:
- * 0 means "copy the respective dimension of the input".
- * i.e., if the input has 2 as its 1st dimension,
- * the output will have 2 as its 1st dimension as well
- * -1 stands for "infer this from the other dimensions"
- * this dimension is calculated to keep the overall element count the same as in the input.
- * At most one -1 can be used in a reshape operation.
+ * Reshape the input tensor with supporting automatic size inference.
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala]: Reshape the input tensor (with automatic size inference support)

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala [hunk 2]
```diff
@@ -23,21 +23,27 @@ import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
...
+ * Positive numbers in the `size` argument are used to reshape the input to the
+ * corresponding dimension size.
+ * There are also two special values allowable in `size`:
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala]: There are also two special values allowed for `size`:

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala (hunk 1 above)
Comment (psyyz10) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala]: Done

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala (hunk 2 above)
Comment (psyyz10) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala]: Done

**Pull request merged**